Header refactoring

lcummings12 opened the pull request:
## Summary

Modularize header files in a way similar to the current configuration of `guide-header.html`, implement a `header-stack.html` component, which consists of the title/deck/kicker.

## Preview

[Link to Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Header-Refactoring/)

## Solution

The solution implements the `header-stack.html` partial, which handle two new parameters `append_community` and `hide_kicker`. The `append_community` parameter appends "Community" to the title, and the `hide_kicker` parameter conditionally hides the kicker text. The community page template was updated to utilize these parameters. These parameters are needed to assure the community template can maintain the unique aspects of its appearance while still adhering to the modular setup of the components.

`header-stack` renders the common elements of the headers, the title, kicker, and deck. Parameters for hiding specific aspects of the component are included to allow for a flexible setup.

The `resources-header` and `communities-header` are modeled after the `guides-header` setup which separates the header from the `single.html` file. Changes consist solely of refactoring and have no change in appearance.

## How To Test
| Description              | Original URL                                                                                   | Updated URL                                                                                                               |
|--------------------------|------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------|
| Communities              | [Original Communities](https://digital.gov/communities/communicators/)                         | [Updated Communities](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Header-Refactoring/communities/communicators/) |
| Event                    | [Original Event](https://digital.gov/event/2024/06/20/uswds-monthly-call-june-2024/)            | [Updated Event](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Header-Refactoring/event/2024/06/20/uswds-monthly-call-june-2024/) |
| Resources                | [Original Resources](https://digital.gov/topics/accessibility/)                                | [Updated Resources](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Header-Refactoring/topics/accessibility/) |
| News                     | [Original News](https://digital.gov/2024/08/01/supporting-innovation-through-the-american-rescue-plan/) | [Updated News](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Header-Refactoring/2024/08/01/supporting-innovation-through-the-american-rescue-plan/) |
| Guides                   | [Original Guides](https://digital.gov/guides/web-analytics-playbook/)                          | [Updated Guides](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/refactor-usa/guides/web-analytics-playbook/) |


Before opening this PR, make sure you’ve done whichever of these applies to you:
- [ ] Branch is up-to-date and includes latest from `main`
- [x] PR has correct labels
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)

nick-mon1 reviewed (changes requested):
### Next Steps

- Use the new `page-header.html` partial on all applicable pages.
- Update the PR description with a list links to updated pages for testing.

lcummings12 commented:
@nick-mon1 
I integrated `page-header` into the ` _default`, `communities`, `events`, `news`, `guides`, and `resources` templates. 

As we discussed last week, removed the redundant *-header components that were needless since they aren't reused, and added the logic directly back into the templates.  I kept the `guide-header` partial as  its used in multiple templates and was originally there.  

I updated table with the most recent version of the pages. 
Thanks!

nick-mon1 reviewed (approved):
@lcummings12 Reverted the code back for guides and resources, they were swapped and displaying incorrectly.
Updated `page-header` to `dg-page-header` for namespacing.

nick-mon1 commented on config/gulp/upload.js:
Can we remove this file from this PR, it is just formatting changes and not related to the work.

nick-mon1 reviewed (approved):
@lcummings12 Thanks for the excellent refactoring. Can we remove old examples or missing new code changes in the PR description? Thanks!
